### PR TITLE
ES-843418-Compare Word documents in DocIO product

### PR DESCRIPTION
**Pull request by Venkateshmuruganandam**
<table>
<tr>
<td>
Task Link
</td>
<td>
<a href="https://dev.azure.com/EssentialStudio/Document%20Processing%20Libraries/_workitems/edit/843418">Add UG example to compare two DOCX format documents</a>
</td>
</tr>
<tr>
<td>
Description
</td>
<td>
Draft UG for Compare Word documents in DocIO product
</td>
</tr>
<tr>
<td>
Content review confirmation
</td>
<td>
completed
</td>
</tr>
</table>

Regards,
Venkatesh

### CODE REVIEW

File: File-Formats-toc.html [hunk 1]
```diff
@@ -203,6 +203,7 @@
 					<li><a href="/file-formats/DocIO/Word-document/Merging-Word-documents">Merging Word documents</a></li>
 					<li><a href="/file-formats/DocIO/Word-document/Print-Word-documents">Print Word documents</a></li>
 					<li><a href="/file-formats/DocIO/Word-document/Split-Word-documents">Split Word documents</a></li>
+					<li><a href="/file-formats/DocIO/Word-document/Compare Word documents">compare Word documents</a></li>
```
Comment (MohanaselvamJothi) [inline File-Formats-toc.html:206]: Compare Word documents

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+---
+
+# Compare Word documents in C#
+
+Comparing two documents allows you to identify differences and similarities between them. The .NET Word library (DocIO) allows you to compare two Word documents and highlight the following changes as tracked changes
+*	Insertions
+*	Deletions
+*	Formatting
+
+N> Comparing two Word documents supported in DOCX format only.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:16]: Add this as 1st  Note

DocIO performs word level comparison while comparing two Word documents.

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/Word-document/Compare Word documents.md (hunk 2 above)
Comment (Venkateshmuruganandam) [reply File-Formats/DocIO/Word-document/Compare Word documents.md]: updated

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+
+{% endhighlight %}
+{% highlight vb.net tabtitle="VB.NET [Windows-specific]" %}
+
+' Loads the original document.
+Using originalDocument As New WordDocument("Data/OriginalDocument.docx", FormatType.Docx)
+    ' Loads the revised document.
+    Using revisedDocument As New WordDocument("Data/RevisedDocument.docx", FormatType.Docx)
+        ' Compare original document and revised documents.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:73]: ' Compare the original and revised documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+End Using
+
+{% endhighlight %}
+{% endtabs %}
+
+![Compare Word documents in C#](WorkingwithWordDocument_images/Compare-Word-documents-without-author.png)
+
+## Set Author and Date 
+
+Compare two Word documents by setting author and date for revisions to identify the changes clearly.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:87]: Compare the two Word documents by setting the author and date for revisions to identify the changes.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+{% tabs %}
+{% highlight c# tabtitle="C# [Cross-platform]" %}
+
+//Loads the original document.
+using (FileStream originalDocumentStreamPath = new FileStream("Data/OriginalDocument.docx", FileMode.Open, FileAccess.Read))
+{
+    using (WordDocument originalDocument = new WordDocument(originalDocumentStreamPath, FormatType.Docx))
+    {
+        //Loads the revised document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:99]: //Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+            }
+        }                 
+    }                           
+}
+
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Loads the original document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:119]: //Load the original document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+using (WordDocument originalDocument = new WordDocument("Data/OriginalDocument.docx", FormatType.Docx))
+{
+    //Loads the revised document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:122]: //Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+    using (WordDocument revisedDocument = new WordDocument("Data/RevisedDocument.docx", FormatType.Docx))
+   {
+        /Compare original document and revised documents.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:125]: /Compare the original and revised documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+}
+
+{% endhighlight %}
+{% highlight vb.net tabtitle="VB.NET [Windows-specific]" %}
+
+' Open the original Word document.
+Using originalDocument As New WordDocument(originalFilePath, FormatType.Docx)
+    ' Open the revised Word document.
+    Using revisedDocument As New WordDocument(revisedFilePath, FormatType.Docx)
+        ' Compare original document with revised document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:139]: ' Compare the original document with the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+End Using
+
+{% endhighlight %}
+{% endtabs %}
+
+![Compare Word documents in C#](WorkingwithWordDocument_images/Compare-Word-documents.png)
+
+## Comparison options
+
+You can customize the Word comparison using our ComparisonOptions in DocIO.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:153]: You can customize the Word comparison using our ComparisonOptions in the DocIO.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+### Ignore format changes
+
+In the .NET Word library (DocIO), document comparison includes formatting changes by default. However, you can configure DocIO to ignore formatting differences using DetectFormatChanges API to concentrate solely on content modifications.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:157]: In the .NET Word library (DocIO), document comparison includes formatting changes by default. However, you can configure DocIO to ignore formatting differences using the DetectFormatChanges API to concentrate solely on content modifications.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+
+The following code example illustrates how to compare two Word documents by ignoring the format changes.
+
+
+{% tabs %}
+{% highlight c# tabtitle="C# [Cross-platform]" %}
+
+//Loads the original document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:165]: //Load the original document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+using (FileStream originalDocumentStreamPath = new FileStream("OriginalDocument.docx", FileMode.Open, FileAccess.Read))
+{
+    using (WordDocument originalDocument = new WordDocument(originalDocumentStreamPath, FormatType.Docx))
+    {
+        //Loads the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:170]: //Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+        using (FileStream revisedDocumentStreamPath = new FileStream("RevisedDocument.docx", FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+        {
+            using (WordDocument revisedDocument = new WordDocument(revisedDocumentStreamPath, FormatType.Automatic))
+            {
+                //Sets the Comparison option detect format changes, whether to detect format changes while comparing two Word documents
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:175]: //Set the Comparison option to detect format changes, whether to detect format changes while comparing two Word documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+                ComparisonOptions compareOptions = new ComparisonOptions();
+                compareOptions.DetectFormatChanges = false;
+                //Compares the original document with revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:178]: //Compare the original document with the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+                originalDocument.Compare(revisedDocument, "Syncfusion", DateTime.Now, compareOptions);
+                //Saves the Word document to MemoryStream
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:180]: //Save the Word document to MemoryStream.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+                }
+            }
+        }
+    }
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Loads the original document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:193]: //Load the original document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 18]
```diff
@@ -0,0 +1,229 @@
...
+    }
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Loads the original document 
+using (WordDocument originalDocument = new WordDocument("OriginalDocument.docx"))
+{
+    //Loads the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:196]: //Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 19]
```diff
@@ -0,0 +1,229 @@
...
+    using (WordDocument revisedDocument = new WordDocument("RevisedDocument.docx"))
+    {
+        //Sets the Comparison option detect format changes, whether to detect format changes while comparing two Word documents
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:199]: //Set the Comparison option detect format changes, whether to detect format changes while comparing two Word documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 20]
```diff
@@ -0,0 +1,229 @@
...
+        ComparisonOptions compareOptions = new ComparisonOptions();
+        compareOptions.DetectFormatChanges = false;
+        //Compare the original document with revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:202]: //Compare the original document with the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 21]
```diff
@@ -0,0 +1,229 @@
...
+        originalDocument.Compare(revisedDocument, "Syncfusion", DateTime.Now, compareOptions);
+        //Save the Word document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:204]: //Save the Word document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 22]
```diff
@@ -0,0 +1,229 @@
...
+        originalDocument.Save(output);
+    }                 
+} 
+
+{% endhighlight %}
+{% highlight vb.net tabtitle="VB.NET [Windows-specific]" %}
+
+'Loads the original document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:212]: 'Load the original document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 23]
```diff
@@ -0,0 +1,229 @@
...
+Using originalDocument As New WordDocument("OriginalDocument.docx")
+    'Loads the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:214]: 'Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 24]
```diff
@@ -0,0 +1,229 @@
...
+    Using revisedDocument As New WordDocument("RevisedDocument.docx")
+        'Sets the Comparison option to detect format changes
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:216]: 'Set the Comparison option to detect format changes.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 25]
```diff
@@ -0,0 +1,229 @@
...
+        Dim compareOptions As New ComparisonOptions()
+        compareOptions.DetectFormatChanges = False
+        'Compares the original document with the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:219]: 'Compare the original document with the revised document.

**Review by florenceliliana — CHANGES_REQUESTED** [review]
The content review is completed.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 26]
```diff
@@ -0,0 +1,231 @@
...
+using (FileStream originalDocumentStreamPath = new FileStream("OriginalDocument.docx", FileMode.Open, FileAccess.Read))
+{
+    using (WordDocument originalDocument = new WordDocument(originalDocumentStreamPath, FormatType.Docx))
+    {
+        //Load the revised document
+        using (FileStream revisedDocumentStreamPath = new FileStream("RevisedDocument.docx", FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+        {
+            using (WordDocument revisedDocument = new WordDocument(revisedDocumentStreamPath, FormatType.Automatic))
+            {
+                //Set the Comparison option to detect format changes, whether to detect format changes while comparing two Word documents.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:173]: //Set the Comparison option to detect format changes while comparing two Word documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 27]
```diff
@@ -0,0 +1,231 @@
...
+                ComparisonOptions compareOptions = new ComparisonOptions();
+                compareOptions.DetectFormatChanges = false;
+                //Compare the original document with the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:176]: //Compare the original document with the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 28]
```diff
@@ -0,0 +1,231 @@
...
+                originalDocument.Compare(revisedDocument, "Syncfusion", DateTime.Now, compareOptions);
+                //Save the Word document to MemoryStream
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:178]: //Save the Word document to MemoryStream.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 29]
```diff
@@ -0,0 +1,231 @@
...
+                }
+            }
+        }
+    }
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Load the original document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:191]: //Load the original document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 30]
```diff
@@ -0,0 +1,231 @@
...
+    }
+}
+
+{% endhighlight %}
+{% highlight c# tabtitle="C# [Windows-specific]" %}
+
+//Load the original document 
+using (WordDocument originalDocument = new WordDocument("OriginalDocument.docx"))
+{
+    //Load the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:194]: //Load the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 31]
```diff
@@ -0,0 +1,231 @@
...
+    using (WordDocument revisedDocument = new WordDocument("RevisedDocument.docx"))
+    {
+        //Set the Comparison option detect format changes, whether to detect format changes while comparing two Word documents.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:197]: //Set the Comparison option to detect format changes while comparing two Word documents.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 32]
```diff
@@ -0,0 +1,231 @@
...
+        ComparisonOptions compareOptions = new ComparisonOptions();
+        compareOptions.DetectFormatChanges = false;
+        //Compare the original document with the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:200]: //Compare the original document with the revised document.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 33]
```diff
@@ -0,0 +1,231 @@
...
+} 
+
+{% endhighlight %}
+{% highlight vb.net tabtitle="VB.NET [Windows-specific]" %}
+
+'Load the original document
+Using originalDocument As New WordDocument("OriginalDocument.docx")
+    'Load the revised document
+    Using revisedDocument As New WordDocument("RevisedDocument.docx")
+        'Set the Comparison option to detect format changes
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:214]: 'Set the Comparison option to detect format changes.

File: File-Formats/DocIO/Word-document/Compare Word documents.md [hunk 34]
```diff
@@ -0,0 +1,231 @@
...
+        Dim compareOptions As New ComparisonOptions()
+        compareOptions.DetectFormatChanges = False
+        'Compare the original document with the revised document
```
Comment (florenceliliana) [inline File-Formats/DocIO/Word-document/Compare Word documents.md:217]: 'Compare the original document with the revised document.

**Review by florenceliliana — COMMENTED** [review]
The content review is completed.

**Pull request merged**